### PR TITLE
adding a github donation page through a lock

### PR DESCRIPTION
**Pull request by julien51**
Hello!

First of all, thanks for your (hard) work!

### Description

As an OpenSource project, I believe MetaMask should have a `FUNDING.yml` file which lets people make donations if they enjoy the software, like we do!

This pull requests adds donation thru an [Unlock](https://unlock-protocol.com/) lock. Locks are smart contract which let people become members by purchasing a key. The key itself is a non fungible token: 🗝 . (you can view the NFT on [OpenSea](https://opensea.io/assets/0x70841c98472553df984aa033f8d3b8c261d17d82/1) for example)

This specific lock is at [`0x70841c98472553df984aa033f8d3b8c261d17d82`](https://etherscan.io/address/0x70841c98472553df984aa033f8d3b8c261d17d82). The current key price is 1 DAI. I currently own the lock but I'll happily transfer its ownership to you if you can send me an address ;) Keys are valid for 1 month and there can be an infinite number of keys.

The pull-request adds 2 files: `FUNDING.yml` which is a github standard file to add a "donation button" with the right links when people click on it and a file which includes the configuration for the donation page.

Here is what it looks like :

image

You can try this on another project which already uses the donation button such as [Unlock](https://github.com/unlock-protocol/unlock/), [MyCrypto](https://github.com/mycryptohq/mycrypto), or [WalletConnect](https://github.com/WalletConnect/walletconnect-monorepo/)...

When/if you merge this you also need to enable donations in the settings (see https://github.com/MetaMask/metamask-extension/settings): 

image

### CODE REVIEW

**Comment by julien51** [comment]
Hello @danfinlay ! I'd love to get your feedback on that one. We blogged about the [motivations behind that PR](https://unlock-protocol.com/blog/unlock-donations/).

One of the next steps that we're thinking of is to create some kind of "shared" lock between many Ethereum OSS projects. The idea is that people can then chose to support a single project, like Ethlint, or the whole ethereum OSS ecosystem. This shared lock would belong to some kind of shared address (or even a DAO) to organize the funds and maybe distribute them to specific projects.

Anyway, my point is that I'd love to kick off a discussion here :)

**Comment by danfinlay** [comment]
Hey @julien51, sorry for being so slow on this. Could you transfer the lock to `meta.mask.eth`? Will that let us update the picture for our lock (fox looking down)? Will that let us update the lock price?

**Comment by whymarrh** [comment]
Would you also be willing to update this PR to resolve the 1 conflict:

```diff
@@ -9,4 +9,4 @@ community_bridge: # Replace with a single Community Bridge project-name e.g., cl
 liberapay: # Replace with a single Liberapay username
 issuehunt: # Replace with a single IssueHunt username
 otechie: # Replace with a single Otechie username
-custom: # Replace with up to 4 custom sponsorship URLs e.g., ['link1', 'link2']
+custom: ['https://donate.unlock-protocol.com/?r=MetaMask/metamask-extension']
```

File: .unlock-protocol.config.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{
+   "icon": "https://assets.unlock-protocol.com/tmp/metamask.png",
+   "persistentCheckout": true,
+   "callToAction": {
+     "default": "Please support work on Metamask by purchasing a key to our Lock on Ethereum!",
```
Comment (whymarrh) [inline .unlock-protocol.config.json]:
```suggestion
     "default": "Please support work on MetaMask by purchasing a key to our Lock on Ethereum!",
```

**Review by whymarrh — CHANGES_REQUESTED** [review]
One other quick comment

**Comment by danfinlay** [comment]
I think @julien51 may be very busy, and we might want to just fork his PR with a fresh lock if we want to add it sooner. I'm okay waiting a bit longer for him to transfer the existing lock if possible.

**Comment by whymarrh** [comment]
> I'm okay waiting a bit longer for him to transfer the existing lock if possible.

Agreed, I'll close this for now and we can reopen this same PR once we've got it sorted out

**Pull request closed without merging**